### PR TITLE
Prepare for 0.6rc1 release

### PR DESCRIPTION
**Pull request by hakonanes**
#### Description of the change

Release prep:
* Bump version to release candidate 0.6rc1
* Add @CSSFrancis to package credits
* Update README to use format in orix
* Update license year to 2024

Other changes:
* Actually remove the class method `StructureLibrary.from_crystal_systems()`. The method previously threw a NotImplementedError, but will now throw an AttributeError. Mentioned in the changelog under *Removed*.
* Replaced deprecated call to `Rotation.from_neo_euler()` with `Rotation.from_axes_angles()`.

I've cleaned up the test suite a bit:
* Replaced most uses of `xfail(error)` with `with pytest.raises(error):`. To me, it seems like best practice is to use the latter when we intentionally raise errors in the code, while the former is mostly used as a temporary measure for bugs that will be fixed.
* Silenced some warnings by making pytest ignoring them
* Close most of "opened" Matplotlib plots in tests using `plt.close()` (silencing a warning of > 20 open plots from Matplotlib)
* Yield a temporary pickle file to tests via fixture, removing the need for manually removing the file in a test

#### Progress of the PR
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Unit tests with pytest for all lines
- [x] Clean code style by [running black](https://diffsims.readthedocs.io/en/latest/contributing.html#get-the-style-right)
- [x] Wait for #209

#### For reviewers

- [ ] The PR title is short, concise, and will make sense 1 year later.
- [ ] New functions are imported in corresponding `__init__.py`.
- [ ] New features, API changes, and deprecations are mentioned in the
      unreleased section in `CHANGELOG.rst`.
- [ ] Contributor(s) are listed correctly in `credits` in `diffsims/release_info.py` and
      in `.zenodo.json`.

### CODE REVIEW

**Comment by hakonanes** [comment]
@CSSFrancis, I listed you as a maintainer alongside Duncan, Phillip, and myself in the release info. Which I thought you already were. I hope this is OK?

**Review by CSSFrancis — APPROVED** [review]
> @CSSFrancis, I listed you as a maintainer alongside Duncan, Phillip, and myself in the release info. Which I thought you already were. I hope this is OK?

That's probably fine.  I haven't really done much with `diffsims` until recently so it makes sense, I was planning on doing some more in the next couple of months. 

As far as the PR goes I looked through all of the changes and it looks good to me

**Comment by hakonanes** [comment]
Great and great!

Thanks for looking it over. We should be good to merge and release once your doc examples PR is in.

**Pull request merged**
